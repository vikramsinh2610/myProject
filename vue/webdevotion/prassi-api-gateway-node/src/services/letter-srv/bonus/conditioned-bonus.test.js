const ConditionedBonus = require('./conditioned-bonus');

describe('ConditionedBonus', () => {
  test('should have these properties', () => {
    const bonus = new ConditionedBonus({
      amount: 1000,
      paymentTime: 'prepayment',
      conditions: [{}, {}],
    });
    expect(bonus).toHaveProperty('amount');
    expect(bonus).toHaveProperty('paymentTime');
    expect(bonus).toHaveProperty('conditions');
    expect(bonus).toHaveProperty('productivePeriodPaymentDelay');
  });

  test('should return it s JSON Schema', () => {
    expect(ConditionedBonus.getJSONSchema()).toMatchSnapshot();
  });
});
