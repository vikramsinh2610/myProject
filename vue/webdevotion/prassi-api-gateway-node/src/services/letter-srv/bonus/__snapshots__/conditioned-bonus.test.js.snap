// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConditionedBonus should return it s JSON Schema 1`] = `
Object {
  "description": "Conditioned Bonus",
  "properties": Object {
    "amount": Object {
      "type": "integer",
    },
    "conditions": Object {
      "items": Object {
        "properties": Object {
          "targets": Object {
            "items": Object {
              "properties": Object {
                "kpi": Object {
                  "properties": Object {
                    "_id": Object {
                      "type": "string",
                    },
                    "options": Object {
                      "additionalProperties": true,
                      "type": "object",
                    },
                    "type": Object {
                      "enum": Array [
                        "number",
                        "currency",
                        "percentage",
                      ],
                      "type": "string",
                    },
                    "value": Object {
                      "type": "integer",
                    },
                  },
                  "required": Array [
                    "_id",
                  ],
                  "type": "object",
                },
                "targetValue": Object {
                  "type": "integer",
                },
                "weightPercentage": Object {
                  "maximum": 10000,
                  "minimum": 0,
                  "type": "integer",
                },
              },
              "required": Array [
                "kpi",
                "targetValue",
                "weightPercentage",
              ],
              "type": "object",
            },
            "type": "array",
          },
        },
        "required": Array [
          "targets",
        ],
        "type": "object",
      },
      "type": "array",
    },
    "maxRecoveryPercentage": Object {
      "maximum": 10000,
      "minimum": 0,
      "type": "integer",
    },
    "paymentTime": Object {
      "enum": Array [
        "prepayment",
        "payment-delayed",
      ],
      "type": "string",
    },
    "productivePeriodPaymentDelay": Object {
      "maximum": 3,
      "minimum": 0,
      "type": "integer",
    },
    "type": Object {
      "enum": Array [
        "conditioned",
      ],
      "type": "string",
    },
  },
  "required": Array [
    "amount",
    "type",
    "paymentTime",
    "conditions",
  ],
  "type": "object",
}
`;
